sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -224,12 +224,12 @@ object DateTimeUtils {
    * value. The return type is [[Option]] in order to distinguish between 0L and null. The following
    * formats are allowed:
    *
-   * `yyyy`
-   * `yyyy-[m]m`
-   * `yyyy-[m]m-[d]d`
-   * `yyyy-[m]m-[d]d `
-   * `yyyy-[m]m-[d]d [h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
-   * `yyyy-[m]m-[d]dT[h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
+   * `[+-]y*`
+   * `[+-]y*-[m]m`
+   * `[+-]y*-[m]m-[d]d`
+   * `[+-]y*-[m]m-[d]d `
+   * `[+-]y*-[m]m-[d]d [h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
+   * `[+-]y*-[m]m-[d]dT[h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
    * `[h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
```

Comment:
for `+12:12:12` do we fail or simply ignore the `+`?

Reply:
we simply ignore the `+`

Reply:
I think failing is better? Since we won't respect `-` here anyway.

Reply:
actually the we will respect the `-`, for example:
```
scala> sql("select date'-12-12-12'").show()
+------------------+
|DATE '-0012-12-12'|
+------------------+
|       -0012-12-12|
+------------------+

scala> sql("select date'+12-12-12'").show()
+-----------------+
|DATE '0012-12-12'|
+-----------------+
|       0012-12-12|
+-----------------+

```

Reply:
are you sure? It seems we only consider the sign for year: https://github.com/apache/spark/pull/32959/files#diff-f59286f6576476c68ae766de0430fc41d424804dbb0658e0dccd998b0d9a756dR404

Reply:
I see. good catch.